src/main/java/doke/TaskList.java

```diff
@@ -0,0 +1,173 @@
+package doke;
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+
+import java.time.format.DateTimeFormatter;
+
+import java.util.ArrayList;
+import java.util.Scanner;
+
+/**
+ * A class to represent a TaskList.
+ */
+public class TaskList {
+
+    private ArrayList<Task> taskList = new ArrayList<>();
+
+    /**
+     * A public constructor for the TaskList class.
+     * It will try to read and create and ArrayList of type task from storage,
+     * which, if not possible would then create an empty ArrayList of type Task.
+     *
+     * @param ui
+     * @param storage
+     */
+    public TaskList(Ui ui, Storage storage) {
+
+        try {
+            Scanner s = new Scanner(Storage.DOKE_FILE);
+            while (s.hasNext()) {
+                String line = s.nextLine();
+                String specialString = " [|] ";
+                String[] temp = line.split(specialString);
+                Task addTask;
+
+                if (temp[0].equals("T")) {
+                    addTask = new ToDo(temp[2]);
+                } else if (temp[0].equals("D")) {
+                    addTask = new Deadline(temp[2], temp[3]);
+                } else {
+                    if (!temp[0].equals("E")) {
+                        continue;
+                    }
+                    addTask = new Events(temp[2], temp[3]);
+                }
+                this.taskList.add(addTask);
+
+                try {
+                    if (temp[1].equals("1")) {
+                        addTask.markDone();
+                    } else {
+                        addTask.markNotDone();
+                    }
+                } catch (DokeException e) {
+                }
+            }
+        } catch (FileNotFoundException e) {
+            try {
+                storage.DOKE_FILE.createNewFile();
+                ui.printOut("a new Doke.txt file has been created"
+                        + "it is in the path mentioned above");
+            } catch (IOException a) {
+                ui.printOut("An error occurred. Try again at another time.");
+            }
+        }
+    }
+
+    public void searchString(String str, Ui ui) {
+        int len = taskList.size();
+        int num = 0;
+        String message = "_________________________ \n";
+        int i = 0;
+        message = "_________________________ \n";
+        while (i < len) {
```

Comment:
It seems like i is incremented at the end of the loop no matter what, perhaps the use of the enhanced for loop will make the purpose of this loop more explicit?